File: go/day-21.go
package main

import "fmt"

type T interface {
	int | string
}

func printArray[t T](arr []t) {
	for _, element := range arr {
		fmt.Println(element)
	}
}

func main() {
	var n int
	var intArr []int

	fmt.Scan(&n)
	for i := 0; i < n; i++ {
		var number int
		fmt.Scan(&number)
		intArr = append(intArr, number)
	}

	var strArr []string
	fmt.Scan(&n)
	for i := 0; i < n; i++ {
		var s string
		fmt.Scan(&s)
		strArr = append(strArr, s)
	}

	printArray(intArr)
	printArray(strArr)
}

File: python/day-21.py
from typing import TypeVar


"""
Input (stdin)
3
1
2
3
2
Hello
World

Expected Output
1
2
3
Hello
World
"""

# ref. https://docs.python.org/3/library/typing.html#typing.TypeVar
T = TypeVar('T') # Can be anything

# only available after python 3.12
class Printer[T]:
    def printArray(self, arr: list[T]):
        for element in arr:
            print(element)

n = int(input())
int_array = []
for _ in range(n):
    int_array.append(int(input().strip()))

n = int(input())
str_array = []
for _ in range(n):
    str_array.append(input().strip())

int_printer = Printer()
int_printer.printArray(int_array)

str_printer = Printer()
str_printer.printArray(str_array)







